libraries/lib-module-manager/ModuleManager.h

```diff
@@ -58,15 +58,28 @@ class Module
    fnModuleDispatch mDispatch;
 };
 
-struct PluginProviderDeleter {
-   void operator ()(PluginProvider *pInterface) const;
-};
+class MODULE_MANAGER_API PluginProviderUniqueHandle final
```

Comment:
I think this could be more simplify defined just as `std::unique_ptr<PluginProvider, PluginProviderDeleter>` and just supply that deleter class

Reply:
I've attempted to replace `safenew` and `delete` with just std::make_unique<>. Though the first commit isn't neccessary part of this PR and I remove it